Add `heading` option to match custom headings

nnmrts opened the pull request:
<!--
Read the [contributing guidelines](https://github.com/remarkjs/.github/blob/main/contributing.md).

We are excited about pull requests, but please try to limit the scope, provide a general description of the changes, and remember, it's up to you to convince us to land it.

If this fixes an open issue, link to it in the following way: `Closes GH-123`.

New features and bug fixes should come with tests.

P.S. have you seen our support and contributing docs?
https://github.com/remarkjs/.github/blob/main/support.md
https://github.com/remarkjs/.github/blob/main/contributing.md
-->

Closes #12.

Most of this is described in the respective issue. Motivation behind this was that I basically have a multilingual readme setup but `remark-license` only works on english readmes. I'm not sure the name of the option is perfect. I'm fine with `match`, `regexp` and `matchHeading`, currently this pull request uses `match`.

nnmrts committed:
Add match option

nnmrts committed:
Add `config.js` as possible fixture config file

nnmrts committed:
Add test fixture for match option

nnmrts committed:
Update documentaion for match option

wooorm commented:
similar feedback as https://github.com/remarkjs/remark-contributors/pull/17#issuecomment-782701549!

nnmrts committed:
Add support for string as the heading option

nnmrts committed:
Update tests for string support

nnmrts committed:
Update documentation

nnmrts committed:
Remove redundant regex wrapping

nnmrts commented:
Renamed it to heading (forgot to mention that in the commit message), added string support and removed the regex wrapping to make it case-insensitive, because `mdast-util-heading-range` already does that.

wooorm commented on index.js:
```diff
@@ -23,10 +23,12 @@ var https = 'https://'
     ignoreFinalDefinitions:
       finals === undefined || finals === null ? true : finals,
-    test: licenseHeadingRegexp
+    test
```
```suggestion
    test: test
```

Just to be careful and not break anyone for now

nnmrts replied:
Alright, committed it!

wooorm reviewed (commented):
Sweet, thanks. One tiny nit!

nnmrts committed:
Update index.js

Co-authored-by: Titus <tituswormer@gmail.com>

wooorm commented:
[Released](https://github.com/remarkjs/remark-license/releases/tag/5.2.0)!
